### ISSUE TITLE
Samples/ directory includes incompatible licenses

### ISSUE THREAD

**Issue by Josaphat** [cpplint/cpplint#134]
This PR https://github.com/cpplint/cpplint/pull/13 pulled in some sample code with licenses incompatible with the overall project license (3-Clause BSD).

This may not be an issue if having it in the samples directory counts as what the GPL calls "mere aggregation," but I'm Not A Lawyer.

One concern is that the codelite sample specifically is GPL licensed, but it is not attributed.

**Comment by tkruse** [comment]
Hi, thanks for notifying me. Not a lawyer here, neither. By curiosity, how did you become aware of this?
Let me think a bit about it. Maybe I'll just download the files on demand every time the tests run.

**Comment by tkruse** [comment]
What do you think of adding attribution like this: https://github.com/cpplint/cpplint/commit/e15cfc3173921446a76625238ab0f676c2ce13bf ?

**Comment by Josaphat** [comment]
I was working on writing an internal BSD port for our company and trying to figure out what to write in the LICENSE field. Not sure what prompted me to look in the samples/ directory but I noticed they were third-party code. I figured I'd just call out all the licenses in the port's Makefile but I noticed there was GPL in there which is incompatible with BSD licenses.

As long as the work is attributed I think that solves it. I think the way it's used counts as mere attribution and doesn't make cpplint a derivative work. So 👍
